src/main/java/org/broadinstitute/consent/http/resources/InstitutionResource.java

```diff
@@ -0,0 +1,117 @@
+    this.institutionUtil = institutionUtil;
+  }
+
+  @GET
+  @Produces("application/json")
+  @PermitAll
+  public Response getInstitutions(@Auth AuthUser authUser) {
+    try{
+      User user = userService.findUserByEmail(authUser.getName());
+      Boolean isAdmin = institutionUtil.checkIfAdmin(user);
+      Gson gson = institutionUtil.getGsonBuilder(isAdmin);
+      List<Institution> institutions = institutionService.findAllInstitutions();
+      return Response.ok().entity(gson.toJson(institutions)).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  };
+
+  @GET
+  @Produces("application/json")
+  @Path("/{id}")
+  @PermitAll
+  public Response getInstitution(@Auth AuthUser authUser, @PathParam("id") String paramId) {
+    try{
+      User user = userService.findUserByEmail(authUser.getName());
+      Boolean isAdmin = institutionUtil.checkIfAdmin(user);
+      Gson gson = institutionUtil.getGsonBuilder(isAdmin);
+      Integer id = Integer.parseInt(paramId);
+      Institution institution = institutionService.findInstitutionById(id);
+      return Response.ok().entity(gson.toJson(institution)).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    } 
+  }
+
+  @POST
+  @Consumes("application/json")
+  @Produces("application/json")
+  @RolesAllowed(ADMIN)
+  public Response createInstitution(@Auth AuthUser authUser, String institution) {
+    try{
+      User user = userService.findUserByEmail(authUser.getName());
+      Institution payload = new Gson().fromJson(institution, Institution.class);
+      Institution newInsitution = institutionService.createInstitution(payload, user.getDacUserId());
+      String jsonResponse = new Gson().toJson(newInsitution);
+      return Response.ok().entity(jsonResponse).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @PUT
+  @Consumes("application/json")
+  @Produces("application/json")
+  @Path("/{id}")
+  @RolesAllowed(ADMIN)
+  public Response updateInstitution(@Auth AuthUser authUser, @PathParam("id") String paramId, String institution) {
+    try{
+      User user = userService.findUserByEmail(authUser.getName());
+      Integer id = Integer.parseInt(paramId);
+      Institution payload = new Gson().fromJson(institution, Institution.class);
+      Institution updatedInstitution = institutionService.updateInstitutionById(payload, id, user.getDacUserId());
+      String jsonResponse = new Gson().toJson(updatedInstitution);
+      return Response.ok().entity(jsonResponse).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @DELETE
+  @Produces("application/json")
+  @Path("/{id}")
+  @RolesAllowed(ADMIN)
+  public Response deleteInstitution(@Auth AuthUser authUser, @PathParam("id") String paramId) {
+    try {
+      Integer id = Integer.parseInt(paramId);
+      institutionService.deleteInstitutionById(id);
```

Comment:
Similar feedback here: use `Integer` in the signature and make sure there really is an institution with the id before trying to delete it.